cmd/common/signer/signer.go

```diff
@@ -94,10 +95,18 @@ func validateEnrollmentCertificate(b []byte) error {
 
 func (si *Signer) Sign(msg []byte) ([]byte, error) {
 	digest := util.ComputeSHA256(msg)
```

Comment:
The digest is always being computed, even though it is only used by one of the switch cases below. It should be moved into that switch case to avoid redundant work